src/DialogFonts.cpp

```diff
@@ -76,9 +76,13 @@ void DialogFonts::updateFont() {
 	font.setPointSize(ui.fontSize->currentData().toInt());
 	QString fontName = font.toString();
 
-	if (document_) {
-		document_->action_Set_Fonts(fontName);
+	const bool applyToAll = ui.fontApplyAll->isChecked();
+
+	if (applyToAll) {
+		for (DocumentWidget *doc : DocumentWidget::allDocuments()) {
+			doc->action_Set_Fonts(fontName);
+		}
 	} else {
-		Preferences::SetPrefFont(fontName);
+		document_->action_Set_Fonts(fontName);
```

Comment:
I think this change here breaks the ability to save the preference :-/

Reply:
This seem more complicated that I thought. Maybe let me clean up the whole thing instead, then resubmit. Sorry for the trouble.

Reply:
No worries, though I think just reverting that one line would have been correct...

Reply:
For reference, many of the config dialogs either take a document or a nullptr, where the intent is to either apply the settings to that specific document, OR if it's nullptr, apply to the default settings.